chore(docs): moved technical to contributor/architecture (#351)
kyma-project/cloud-manager: docs/contributor/architecture/README.md

# SKR Cloud Resources Controller Manager

Due to the non-scalable concurrent reconciliation of large number of clusters the SKR Cloud Resources Controller Manager 
cannot keep long-living connections on the remote clusters permanently watching for changes. Instead, a custom
SKR Looper component loops through SKRs with the Cloud Manager module enabled, and instantiates new 
ControllerRuntime manager that lists all the "watched" (reconciler registered with the manager with `.For()` or `.Watches()` 
methods as defined in controller-runtime) and with them maintain a shot lived "cache" until all SKR reconcilers are 
called with respective resources they are managing. Once all is done, all resources for that SKR, shot-living cache, client... 
are disposed and the same process is repeated for the next SKR. 

The reconciler facing API like `Reconcile()` and `.SetupWithManager()` functions remains as close as possible to 
the one defined by controller-runtime and used by Kubebuilder. 

> ![SKR Controller Manager](./assets/skr-controller-manager.drawio.svg)

Reviewer: The extension in the source file needs also to be changed - they should match.

## CloudControl Scope resource

Different cloud providers APIs require different connection options to define the scope of the operations:
* GCP - project
* Azure - tenant and subscription
* AWS - account

The SKR loop, when projecting into the KCP resources, will set the `kymaName` field as the indication which SKR that 
resources refers two. So, the starting point for the KCP loop reconcilers is the Kyma CR name. Following the path from 

> Kyma CR in KCP to the Shoot CR in Garden it is possible to determine the cloud provider scope. 

Reviewer: ```suggestion
Kyma CR in KCP to the Shoot CR in Gardener it is possible to determine the cloud provider scope. 
```
Garden or Gardener?

The Cloud Control KCP resource Scope was introduced to 

> * improve performance by not reading so many different resources and reaching out to the Garden

Reviewer: ```suggestion
* improve performance by not reading so many different resources and reaching out to Gardener
```
Gardener?

> * simply the development landscape by not avoiding the necessity to beside local KCP cluster also have the Garden cluster 

Reviewer: ```suggestion
* simplify the development landscape by not avoiding the necessity to beside the local KCP cluster and also have the Gardener cluster 
```
I'm not sure if I follow: "by not avoiding the necessity to beside the local KCP cluster". Can we rephrase?

  with all the relevant resources

> Upon the appearance of the first resource from certain SKR the cloud provider scope is determined, saved in the 

Reviewer: ```suggestion
Upon the appearance of the first resource from a certain SKR the cloud provider scope is determined, saved in the 
```

> Scope KCP resource, and original Cloud Control KCP resource updated with the `scopeRef`. In dev, scope determination 

Reviewer: ```suggestion
Scope KCP resource and the original Cloud Control KCP resource is updated with the `scopeRef`. In dev, scope determination 
```

can be avoided by creating a resource already referring to an existing Scope resource. 

## IpRange

> Some cloud resources require an allocation of the private IP, like NFS instance. Since network security is managed 

Reviewer: ```suggestion
Some cloud resources require an allocation of a private IP, like an NFS instance. Since network security is managed 
```
